test/Utils/UtilsParser.test.js

```diff
@@ -62,4 +62,81 @@ describe('#utilsParser', () => {
         });
     });
   });
+  describe('referenceData()', () => {
+    it('should parse sample data response for Country', () => {
+      const uParser = new Parser('util:ReferenceDataRetrieveRsp', 'v47_0', {});
+      const parseFunction = utilsParser.REFERENCE_DATATYPE;
+      const xml = fs.readFileSync(`${xmlFolder}/country-data.xml`).toString();
+      return uParser.parse(xml).then((json) => {
+        const result = parseFunction.call(uParser, json);
+        expect(result).to.be.an('array').and.to.have.lengthOf(3);
+        result.forEach(c => expect(c).to.be.an('object').and.to.have.all.keys(['code', 'name', 'deprecated']));
+      });
+    });
+
+    it('should parse sample data response for Hotel Amenities', () => {
+      const uParser = new Parser('util:ReferenceDataRetrieveRsp', 'v47_0', {});
+      const parseFunction = utilsParser.REFERENCE_DATATYPE;
+      const xml = fs.readFileSync(`${xmlFolder}/hotelAmenities-data.xml`).toString();
+      return uParser.parse(xml).then((json) => {
+        const result = parseFunction.call(uParser, json);
+        expect(result).to.be.an('array').and.to.have.lengthOf(3);
+        result.forEach(c => expect(c).to.be.an('object').and.to.have.all.keys(['code', 'name', 'description', 'deprecated', 'additionalInfo']));
+      });
+    });
+
+    it('should parse sample data response for Payment Type', () => {
+      const uParser = new Parser('util:ReferenceDataRetrieveRsp', 'v47_0', {});
+      const parseFunction = utilsParser.REFERENCE_DATATYPE;
+      const xml = fs.readFileSync(`${xmlFolder}/paymentType-data.xml`).toString();
+      return uParser.parse(xml).then((json) => {
+        const result = parseFunction.call(uParser, json);
+        expect(result).to.be.an('array').and.to.have.lengthOf(3);
+        result.forEach(c => expect(c).to.be.an('object').and.to.have.all.keys(['code', 'name', 'description', 'deprecated', 'additionalInfo']));
+      });
+    });
+
+    it('should throw parsing error', () => {
+      const parseFunction = utilsParser.REFERENCE_DATATYPE;
+      try {
+        parseFunction({});
```

Comment:
If the function doesn't throw, then the test will be green

Reply:
Did not get what you meant. Actually it does throw since null object does not have any more properties, and that is checked in line 27 of src/Services/Utils/UtilsParser.js

Reply:
;) hmm.. made some changes.. should.fail added to the end after try..catch